### PR TITLE
Allow sending messages in a background thread

### PR DESCRIPTION
**Pull request by acasajus**
Allow sending messages from a background thread when enabled.

### CODE REVIEW

**Review by nguyenkims — APPROVED** [review]
LGTM, I just left a suggestion on the variable name.

File: app/mail_sender.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    POSTFIX_SERVER,
+)
+from app.email import headers
+from app.log import LOG
+from app.message_utils import message_to_bytes
+
+
+@dataclass
+class SendRequest:
+    from_address: str
```
[the remark this answers is not in the document]
Comment (acasajus) [reply app/mail_sender.py:23]: Changed

File: app/mail_sender.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    to_address: str
```
[the remark this answers is not in the document]
Comment (acasajus) [reply app/mail_sender.py]: changed

File: app/mail_sender.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
+import time
+from concurrent.futures import ThreadPoolExecutor
+from mailbox import Message
+from smtplib import SMTP, SMTPServerDisconnected, SMTPRecipientsRefused
+from typing import Optional, Dict
+
+import newrelic
```
[the remark this answers is not in the document]
Comment (acasajus) [reply app/mail_sender.py]: Good catch :ok_hand:

**Pull request merged**